Author: Ignaz Forster <iforster@suse.com>
Date:   Wed Jan 14 17:57:52 2020 +0100
Upstream: Ticket opened [gh#coreos/ignition#890]

    Implement poor man's solution for mounting a device multiple times,
    e.g. to mount several subvolumes from a Btrfs device or bind mounting
    the device to multiple places, by also adding the path to the key.

Index: ignition-2.3.0/config/v3_1/types/filesystem.go
===================================================================
--- ignition-2.3.0.orig/config/v3_1/types/filesystem.go
+++ ignition-2.3.0/config/v3_1/types/filesystem.go
@@ -23,7 +23,7 @@ import (
 )
 
 func (f Filesystem) Key() string {
-	return f.Device
+	return f.Device + *f.Path
 }
 
 func (f Filesystem) IgnoreDuplicates() map[string]struct{} {
Index: ignition-2.3.0/config/v3_2_experimental/types/filesystem.go
===================================================================
--- ignition-2.3.0.orig/config/v3_2_experimental/types/filesystem.go
+++ ignition-2.3.0/config/v3_2_experimental/types/filesystem.go
@@ -23,7 +23,7 @@ import (
 )
 
 func (f Filesystem) Key() string {
-	return f.Device
+	return f.Device + *f.Path
 }
 
 func (f Filesystem) IgnoreDuplicates() map[string]struct{} {
